src/extensions/displayExtensionInfo.ts

```diff
@@ -1,37 +1,233 @@
-  if (source.spec.description) {
-    lines.push(`**Description**: ${source.spec.description}`);
+  lines.push(`**Author**: ${spec.author?.authorName} ${urlMarkdown}`);
+  if (spec.description) {
+    lines.push(`**Description**: ${spec.description}`);
+  }
+  if (published) {
+    if (spec.license) {
+      lines.push(`**License**: ${spec.license}`);
+    }
+    lines.push(`**Source code**: ${spec.sourceUrl}`);
   }
   if (lines.length > 0) {
-    utils.logLabeledBullet(logPrefix, `information about ${extensionName}:`);
+    utils.logLabeledBullet(logPrefix, `information about '${clc.bold(extensionName)}':`);
     const infoStr = lines.join("\n");
     // Convert to markdown and convert any trailing newlines to a single newline.
     const formatted = marked(infoStr).replace(/\n+$/, "\n");
     logger.info(formatted);
+    // Return for testing purposes.
+    return lines;
   } else {
     throw new FirebaseError(
       "Error occurred during installation: cannot parse info from source spec",
       {
         context: {
-          source: source,
+          spec: spec,
           extensionName: extensionName,
         },
       }
     );
   }
 }
+
+/**
+ * Prints out all changes to the spec that don't require explicit approval or input
+ *
+ * @param spec The current spec of a ExtensionInstance
+ * @param newSpec The spec that the ExtensionInstance is being updated to
+ * @param published whether or not this spec is for a published extension
+ */
+export function displayUpdateChangesNoInput(
+  spec: extensionsApi.ExtensionSpec,
+  newSpec: extensionsApi.ExtensionSpec,
+  published = false
+): string[] {
+  const lines: string[] = [];
+  if (spec.displayName !== newSpec.displayName) {
+    lines.push(
+      "",
+      "**Name:**",
+      deletionColor(`- ${spec.displayName}`),
+      additionColor(`+ ${newSpec.displayName}`)
+    );
+  }
+
+  if (spec.author?.authorName !== newSpec.author?.authorName) {
+    lines.push(
+      "",
+      "**Author:**",
+      deletionColor(`- ${spec.author?.authorName}`),
+      additionColor(`+ ${spec.author?.authorName}`)
+    );
+  }
+
+  if (spec.description !== newSpec.description) {
+    lines.push(
+      "",
+      "**Description:**",
+      deletionColor(`- ${spec.description}`),
+      additionColor(`+ ${newSpec.description}`)
+    );
+  }
+
+  if (published) {
+    if (spec.sourceUrl !== newSpec.sourceUrl) {
+      lines.push(
+        "",
+        "**Source code:**",
+        deletionColor(`- ${spec.sourceUrl}`),
+        additionColor(`+ ${newSpec.sourceUrl}`)
+      );
+    }
+  }
+
+  if (spec.billingRequired && !newSpec.billingRequired) {
+    lines.push("", "**Billing is no longer required for this extension.**");
+  }
+  logger.info(marked(lines.join("\n")));
+  return lines;
+}
+
+/**
+ * Checks for spec changes that require explicit user consent,
+ * and individually prompts the user for each changed field
+ *
+ * @param spec The current spec of a ExtensionInstance
```

Comment:
Missing a period here and on some other jsdocs descriptions

Reply:
I went thru the code and added all periods that I personally noticed were missing.